rust/shalom_dart_codegen/src/lib.rs

```diff
@@ -70,30 +71,43 @@ mod ext_jinja_fns {
     }
 
     #[allow(unused_variables)]
-    pub fn type_name_for_variable(
+    pub fn type_name_for_input(
         schema_ctx: &SchemaContext,
-        variable: ViaDeserialize<VariableDefinition>,
+        input: ViaDeserialize<InputFieldDefinition>,
     ) -> String {
-        let ty_name = variable.0.ty.name();
-        let resolved = DEFAULT_SCALARS_MAP.get(&ty_name).unwrap();
-        if variable.is_optional && variable.default_value.is_none() {
+        let ty_name = input.0.ty.name();
+        let ty = input.resolve_type(schema_ctx);
+        let resolved = match ty {
+            GraphQLAny::Scalar(_) => DEFAULT_SCALARS_MAP.get(&ty_name).unwrap().clone(),
+            GraphQLAny::InputObject(_) => ty_name,
+            _ => unimplemented!("input type not supported"),
+        };
+        if input.is_optional && input.default_value.is_none() {
             format!("Option<{}?>", resolved)
-        } else if variable.is_optional {
+        } else if input.is_optional {
             format!("{}?", resolved)
         } else {
-            resolved.clone()
+            resolved
         }
     }
 
-    pub fn parse_default_value(variable: ViaDeserialize<VariableDefinition>) -> String {
-        let default_value = variable.0.default_value;
+    pub fn parse_input_default_value(input: ViaDeserialize<InputFieldDefinition>) -> String {
+        let default_value = input.0.default_value;
         if default_value.is_none() {
             panic!("cannot parse default value that does not exist")
         }
         let default_value = default_value.unwrap();
         default_value.to_string()
     }
 
+    pub fn is_field_object(
+        schema_ctx: &SchemaContext,
+        input: ViaDeserialize<InputFieldDefinition>,
+    ) -> bool {
+        let ty = input.resolve_type(schema_ctx);
+        matches!(ty, GraphQLAny::InputObject(_))
+    }
```

Comment:
just do `is_input_type(name: String)` and do in jinja `field.ty` or something like this,